app/services/foods_servicer.rb

```diff
@@ -0,0 +1,18 @@
+class FoodsServicer
+  attr_reader :client
+
+  def initialize(client = FoodsClient)
+    @client = client
+  end
+
+  def search(food)
+    results = client.search_foods(food)[:foods].map do |data|
+      Food.new(data)
+    end
+    results[0..9]
+  end
+
+  def get_total_hits(food)
+    client.search_foods(food)[:totalHits]
+  end
```

Comment:
Ultimately these methods are hitting the same endpoint twice. Instead you could refactor this to something like:
```ruby
  def search(food)
    results = client.search_foods(food)
    foods = results[:foods].map do |data|
      Food.new(data)
    end
    [results[:totalHits], foods[0..9]]
  end
  ```
  
  And then in your controller:
  `    @results, @foods = FoodsServicer.new.search(params[:q])`
  
  This should allow you to get rid of the `get_total_hits` method altogether.